moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,467 @@
+	{
+		for (builtin cell_value : b)
+			cellSeq.push_back(mk_boolean_cell_exec(cell_value, row_num,
+			                                       labels[cell_number++]));
+
+		return cellSeq;
+	}
+
+	HandleSeq operator()(const contin_seq& c)
+	{
+		for (contin_t cell_value : c)
+			cellSeq.push_back(mk_real_cell_exec(cell_value, row_num,
+			                                    labels[cell_number++]));
+
+		return cellSeq;
+	}
+
+	template<typename T>
+	HandleSeq operator()(const T& t) const
+	{}
+
+	int cell_number = 0;
+	HandleSeq cellSeq;
+	int row_num;
+	vector<string> labels;
+};
+
+/**
+ * Represent a row as a list of cells.
+ *
+ * @param cells cells in a row
+ * @return
+ */
+Handle mk_row(const HandleSeq& cells)
+{
+	return createLink(cells, LIST_LINK);
+}
+
+/**
+ * Represent a row as a list of row number and cells
+ *
+ * @param cells cells in a row
+ * @param row_num row number
+ * @return
+ */
+Handle mk_numbered_row(const HandleSeq& cells, const int row_num)
+{
+	return createLink(LIST_LINK,
+	                  mk_row_number_cell(row_num),
+	                  createLink(cells, LIST_LINK));
+}
+
+/**
+ * Create a table of dependent (output) data.
+ *
+ * @param oTable output table
+ * @return
+ */
+Handle mk_output_table(const OTable& oTable)
+{
+	const string& label = oTable.get_label();
+	HandleSeq rows;
+	int row_num = 0;
+
+	for (const vertex& v : oTable)
+	{
+		Handle h = createLink(LIST_LINK,
+		                      mk_row_number_cell(row_num++),
+		                      boost::apply_visitor(mk_cell_visitor(), v));
+		rows.push_back(h);
+	}
+
+	return createLink(SIMILARITY_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  createLink(rows, SET_LINK));
+}
+
+/**
+ * Create a table of independent (input) data.
+ *
+ * @param iTable input table
+ * @todo this method doesn't support a table of mixed types. It assumes that
+ * 		 all features have similar type with the output feature.
+ * @return
+ */
+Handle mk_input_table(const ITable& iTable)
+{
+	const vector<string> labels = iTable.get_labels();
+	HandleSeq rows;
+	int row_num = 0;
+
+	for (const multi_type_seq& m : iTable)
+	{
+		HandleSeq cells = boost::apply_visitor(mk_cell_seq_visitor(),
+		                                       m.get_variant());
+
+		Handle h = createLink(LIST_LINK,
+		                      mk_row_number_cell(row_num++),
+		                      createLink(cells, LIST_LINK));
+		rows.push_back(h);
+	}
+
+	return createLink(SIMILARITY_LINK, parse_header(labels),
+	                  createLink(rows, SET_LINK));
+}
+
+/**
+ * Represent the table using either similarity or list link.
+ *
+ * @param table
+ * @param similarity use similarity link if true
+ * @return
+ */
+HandleSeq mk_full_table(const Table& table, const bool similarity=false)
+{
+	HandleSeq rows;
+	int row_num = 0;
+
+	for (const multi_type_seq& m : table.itable)
+	{
+		HandleSeq cells = boost::apply_visitor(mk_cell_seq_visitor(),
```

Comment:
BTW, it terms of performance, it might be better to create the visitor out of the loop. But frankly it's not even sure as both the compiler and system (CPU branch prediction) optimize that sort of things. Just keep that in mind for the day you want to speed up that function (although it's unlikely it'll be problematic).